src/flb_lib.c

```diff
@@ -668,6 +668,8 @@ int flb_start(flb_ctx_t *ctx)
         fd = event->fd;
         bytes = flb_pipe_r(fd, &val, sizeof(uint64_t));
         if (bytes <= 0) {
+            pthread_cancel(tid);
```

Comment:
Should we need to invoke pthread_cancel ?

Reply:
If we fail to read from the pipe here, then we can't infer the state of the other thread with 100% certainty. So I figured it made sense to cancel it, or else `pthread_join` might block forever. This case should be extremely rare anyway.

Reply:
Thank you. I got it.

> This case should be extremely rare anyway.

I think so.